lib/kubernetes-deploy/kubernetes_resource/horizontalpodautoscaler.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+module KubernetesDeploy
+  class HorizontalPodAutoscaler < KubernetesResource
```

Comment:
The HPA was previously defined only as a string in the `prune` list, we now need an actual class backing it.